Sources/KlaviyoLocation/KlaviyoLocationManager.swift

```diff
@@ -12,132 +12,101 @@ import KlaviyoCore
 import KlaviyoSwift
 import OSLog
 
-class KlaviyoLocationManager: NSObject {
+class KlaviyoLocationManager: NSObject, CLLocationManagerDelegate {
     static let shared = KlaviyoLocationManager()
 
     private var locationManager: LocationManagerProtocol
-    private let geofenceManager: KlaviyoGeofenceManager
 
-    init(locationManager: LocationManagerProtocol? = nil, geofenceManager: KlaviyoGeofenceManager? = nil) {
+    init(locationManager: LocationManagerProtocol? = nil) {
         self.locationManager = locationManager ?? CLLocationManager()
-        self.geofenceManager = geofenceManager ?? KlaviyoGeofenceManager(locationManager: self.locationManager)
 
         super.init()
         self.locationManager.delegate = self
         self.locationManager.allowsBackgroundLocationUpdates = true
         self.locationManager.startMonitoringSignificantLocationChanges()
     }
 
+    // Q: why do we need this?
     deinit {
         locationManager.delegate = nil
         locationManager.stopUpdatingLocation()
         locationManager.stopMonitoringSignificantLocationChanges()
-        geofenceManager.destroyGeofencing()
+        stopGeofenceMonitoring()
     }
 
     @MainActor
-    func setupGeofencing() {
-        if environment.getLocationAuthorizationStatus() == .authorizedAlways {
-            geofenceManager.setupGeofencing()
-        }
-    }
-
-    @MainActor
-    func destroyGeofencing() {
-        geofenceManager.destroyGeofencing()
-    }
-}
-
-extension KlaviyoLocationManager: CLLocationManagerDelegate {
-    // MARK: Authorization
-
-    public func locationManager(_ manager: CLLocationManager, didFailWithError error: Error) {
-        if #available(iOS 14.0, *) {
-            Logger.geoservices.error("Core Location services error: \(error.localizedDescription)")
+    func startGeofenceMonitoring() {
```

Comment:
`setupGeofencing` -> `startGeofenceMonitoring` 

suggesting this name change to be specific about what the method does.